lib/package.py

```diff
@@ -118,9 +117,11 @@ def download_files(self, recurse=True):
         Download package source code and build files.
         Optionally, do the same for its dependencies, recursively.
         """
-        # Download all package sources
-        download_f = partial(package_source.download, directory=PACKAGES_REPOS_TARGET_PATH, local_copy_subdir_name=self.name)
-        self.sources = map(download_f, self.sources)
+        update_packages_repos = CONF.get('build_packages').get('update_packages_repos_before_build')
+        for source in self.sources:
+            if ['url'] not in source.keys() and update_packages_repos:
```

Comment:
The file downloaded from a URL source is not cached in our build system, so it must always be downloaded. The logic was wrong, I fixed it.